### ISSUE TITLE
Broken link in Javadoc

### ISSUE THREAD

**Issue by cowwoc** [google/guava#5457]
1. Go to https://guava.dev/releases/30.1.1-jre/api/docs/
2. Search for "unsignedbytes"
3. Get redirected to https://guava.dev/releases/30.1.1-jre/api/docs/undefined/com/google/common/primitives/UnsignedBytes.html which returns HTTP 404.

**Comment by cpovirk** [comment]
Thanks. We'll need to do the same thing as https://github.com/google/truth/issues/670 (which might also be what we did for https://github.com/google/error-prone/issues/1825, but I haven't looked).

**Comment by cpovirk** [comment]
(As noted on the Truth issue, you might like an alternative that we added, which is to navigate to [guava.dev/unsignedbytes](http://guava.dev/unsignedbytes). But we'll fix search, too.)

**Comment by cowwoc** [comment]
@cpovirk I looked at the issue you linked to but it sounds like everyone is implementing a workaround instead of fixing the underlying problem. Did anyone file a bug report against the Javadoc tool?

**Comment by cowwoc** [comment]
@cpovirk Good find. I'll add it to the Stackoverflow discussion.

**Comment by ddadon10** [comment]
Hi, any news about this issue? I use Guava's Javadoc often and this bug is very annoying.

Thanks :)

**Comment by cpovirk** [comment]
RE: Maven Toolchains: https://github.com/google/error-prone/issues/3895#issuecomment-1542294317

**Comment by cpovirk** [comment]
I was just messing with toolchains a little, and I found that putting this in `maven-compiler-plugin` appears to make no difference:

```
            <jdkToolchain>
              <version>22</version>
            </jdkToolchain>
```

I do see some effect if I configure `maven-toolchains-plugin` (as follows in `pluginManagement`, then listed in `plugins` itself):

```xml
        <plugin>
          <artifactId>maven-toolchains-plugin</artifactId>
          <version>3.2.0</version>
          <executions>
            <execution>
              <goals>
                <goal>toolchain</goal>
              </goals>
            </execution>
          </executions>
          <configuration>
            <toolchains>
              <jdk>
                <version>11</version>
              </jdk>
            </toolchains>
          </configuration>
        </plugin>
```

But then we have the problem that we have to tell Maven where to find the toolchain.

I'm able to autodownload a couple with:

```xml
        <plugin>
          <groupId>org.mvnsearch</groupId>
          <artifactId>toolchains-maven-plugin</artifactId>
          <version>4.5.0</version>
          <executions>
            <execution>
              <goals>
                <goal>toolchain</goal>
              </goals>
            </execution>
          </executions>
          <configuration>
            <toolchains>
              <jdk>
                <version>11</version>
              </jdk>
              <jdk>
                <version>22</version>
              </jdk>
            </toolchains>
          </configuration>
        </plugin>
```

_Then_ the request for 22 appears to take effect.

The configuration above presumably [sets up the _tests_ to run with 11, too](https://maven.apache.org/plugins/maven-compiler-plugin/examples/compile-using-different-jdk.html#:~:text=1%20default%20jdk%20toolchain%20for%20all%20related%20maven%2Dplugins), and that's not what we want (well, at least until we both set up the tests to run with 11 _and_ set up the tests to run with 8 and probably other versions). But luckily, it's not working?

```
[INFO] Toolchain in surefire-plugin: JDK[/usr/local/google/home/cpovirk/.m2/jdks/openjdk-11.0.16_8]
[WARNING] Toolchains are ignored, 'executable' parameter is set to /usr/local/google/home/cpovirk/jdk-17.0.2/bin/java
```

Oh, but maybe it starts working if I upgrade Surefire to 3.3.1? So I should probably figure this out for real.

**Comment by cpovirk** [comment]
I'm now this far:

```patch
diff --git a/guava/pom.xml b/guava/pom.xml
index be0bab174e..cd2c7412da 100644
--- a/guava/pom.xml
+++ b/guava/pom.xml
@@ -99,7 +99,11 @@
         </configuration>
       </plugin>
       <plugin>
-        <artifactId>maven-compiler-plugin</artifactId>
+        <groupId>org.mvnsearch</groupId>
+        <artifactId>toolchains-maven-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <artifactId>maven-toolchains-plugin</artifactId>
       </plugin>
       <plugin>
         <artifactId>maven-source-plugin</artifactId>
diff --git a/pom.xml b/pom.xml
index 7e634f7a64..ea4bba4af0 100644
--- a/pom.xml
+++ b/pom.xml
@@ -12,6 +12,7 @@
   <description>Parent for guava artifacts</description>
   <url>https://github.com/google/guava</url>
   <properties>
+    <surefire.toolchain.version>${java.specification.version}</surefire.toolchain.version>
     <!-- Override this with -Dtest.include="**/SomeTest.java" on the CLI -->
     <test.include>%regex[.*.class]</test.include>
     <truth.version>1.4.4</truth.version>
@@ -123,8 +124,11 @@
       <plugins>
         <plugin>
           <artifactId>maven-compiler-plugin</artifactId>
-          <version>3.8.1</version>
+          <version>3.13.0</version>
           <configuration>
+            <jdkToolchain>
+              <version>22</version>
+            </jdkToolchain>
             <source>1.8</source>
             <target>1.8</target>
             <encoding>UTF-8</encoding>
@@ -158,6 +162,50 @@
             <fork>true</fork>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.mvnsearch</groupId>
+          <artifactId>toolchains-maven-plugin</artifactId>
+          <version>4.5.0</version>
+          <executions>
+            <execution>
+              <goals>
+                <goal>toolchain</goal>
+              </goals>
+            </execution>
+          </executions>
+          <configuration>
+            <toolchains>
+              <jdk>
+                <version>11</version>
+              </jdk>
+              <jdk>
+                <version>22</version>
+              </jdk>
+              <testJdk>
+                <version>${surefire.toolchain.version}</version>
+              </testJdk>
+            </toolchains>
+          </configuration>
+        </plugin>
+        <plugin>
+          <artifactId>maven-toolchains-plugin</artifactId>
+          <version>3.2.0</version>
+          <executions>
+            <execution>
+              <goals>
+                <goal>toolchain</goal>
+              </goals>
+            </execution>
+          </executions>
+          <configuration>
+            <toolchains>
+              <jdk>
+                <!-- TODO: b/286965322 - Use a newer version if it doesn't break JDiff (or Javadoc or whatever). -->
+                <version>11</version>
+              </jdk>
+            </toolchains>
+          </configuration>
+        </plugin>
         <plugin>
           <artifactId>maven-jar-plugin</artifactId>
           <version>3.2.0</version>
@@ -177,7 +225,7 @@
             <dependency>
               <groupId>org.codehaus.plexus</groupId>
               <artifactId>plexus-io</artifactId>
-              <!-- DO NOT UPGRADE this past 3.4.1 until https://github.com/codehaus-plexus/plexus-io/issues/109 is fixed. -->
+              <!-- DO NOT UPGRADE this past 3.4.1 until https://github.com/codehaus-plexus/plexus-io/issues/109 is fixed (probably in 3.5.1). -->
               <version>3.4.1</version>
             </dependency>
           </dependencies>
@@ -246,8 +294,11 @@
         </plugin>
         <plugin>
           <artifactId>maven-surefire-plugin</artifactId>
-          <version>2.7.2</version>
+          <version>3.3.1</version>
           <configuration>
+            <jdkToolchain>
+              <version>${surefire.toolchain.version}</version>
+            </jdkToolchain>
             <includes>
               <include>${test.include}</include>
             </includes>
```

But that fails during `maven-compiler-plugin` (with no error message) when I run with `JAVA_HOME` pointing to openjdk-8u342. It also fails if I dial back the `maven-compiler-plugin` toolchain to 21 or even all the way to 12.

We could [require a newer version to build](https://github.com/google/guava/issues/6549)....

**Comment by cpovirk** [comment]
Ah, the failure is probably from the `maven-compiler-plugin` `<profile>` we have....
